contracts/interfaces/IStakingNFTDescriptor.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >=0.5.0;
+
+  struct DepositExpiryInfo {
+    uint256 trancheId;
+    uint256 stake;
+  }
+
+  struct StakingTokenURIParams {
+    uint256 tokenId;
+    uint256 poolId;
+    address stakingPool;
+    address owner;
+    string name;
+  }
+
+interface IStakingNFTDescriptor {
+
+  function tokenURI(StakingTokenURIParams calldata params) external view returns (string memory);
```

Comment:
Let's avoid the struct, pass the variables directly.

Reply:
stack is pretty deep. whats the reason to pass them directly?